tests/acceptance/30_custom_promise_types/multiline_insert.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/python3
+#
+# Sample custom promise type, uses cfengine.py library located in same dir.
+#
+# Use it in the policy like this:
+# promise agent multiline_insert
+# {
+#     interpreter => "/usr/bin/python3";
+#     path => "$(sys.inputdir)/multiline_insert.py";
+# }
+# bundle agent main
+# {
+#   multiline_insert:
+#     "/home/vagrant/output"
+#       lines => { "line1", "line2", "line3" };
+# }
+
+import os.path
+
+from cfengine import PromiseModule, ValidationError, Result
+
+
+def _file_content_is(fpath, contents):
+    if not os.path.exists(fpath):
+        return False
+
+    try:
+        with open(fpath, "r") as f:
+            return f.read() == contents
+    except:
+        return False
+
+
+class MultilineInsertPromiseTypeModule(PromiseModule):
+    def __init__(self):
+        super().__init__("multiline_insert_promise_module", "0.0.1")
+
+    def validate_promise(self, promiser, attributes):
+        if "lines" not in attributes or type(attributes["lines"]) not in (list, tuple):
+            raise ValidationError("'lines' attribute of type slist required")
```

Comment:
Could also iterate through `attributes["lines"]` and check that all of the items are strings.